orderer/common/cluster/block_fetcher.go

```diff
@@ -505,7 +514,7 @@ func (bf BlockFetcher) HeightsByEndpoints() (map[string]uint64, error) {
 }
 
 // UpdateEndpoints assigns the new endpoints.
-func (p *BlockFetcher) updateEndpoints(endpoints []EndpointCriteria) {
+func (p *BlockFetcher) UpdateEndpoints(endpoints []EndpointCriteria) {
```

Comment:
why did you export this? The whole point in un-exporting it, was that we do it on our own :-)

Reply:
We return an interface [here](https://github.com/shivdeep-singh-ibm/fabric-collab/blob/b854da42c0d7177e9f0639fef1dda4450d5d7189/orderer/common/follower/block_puller.go#L194). To satisfy this interface we need `UpdateEndpoints`, otherwise it will give compilation error.

Reply:
So just don't return an interface, what's the problem?

Reply:
`UpdateEndpoints` is used [here.](https://github.com/shivdeep-singh-ibm/fabric-collab/blob/b854da42c0d7177e9f0639fef1dda4450d5d7189/orderer/common/follower/follower_chain.go#L543) If we are going to use a bft `BlockFetcher` instead of. a `BlockPuller`, it needs to expose `UpdateEndpoints` method.

Reply:
I understand it's used there. It's also used in other places. I'm saying that if this new block pulling unit updates its endpoints autonomously, and its goal is to substitute the block pulling functionality in all places, then there won't be a need to call `UpdateEndpoints` any longer when it is wired instead of the other block puller types.

Reply:
agree. I have kept this for now, for support a successful build and not breaking compilation. So we need to refactor some code to remove the usage of `UpdateEndpoints`, till then we can keep it.

Reply:
Then please remove all uses of the block fetcher in the rest of Fabric, so we could then "move" the block fetcher only into the main Fabric.  Afterwards, we can make PRs that will replace the usage of the block puller or any equivalent primitive with this block fetcher. 

The reason is that then we could simply make a PR that adds the block fetcher into Fabric, and then just cherry pick the commits that replace the primitives with the block fetcher.